synapse/storage/databases/main/event_federation.py

```diff
@@ -1123,10 +1147,16 @@ async def get_backfill_events(
         )
         events = await self.get_events_as_list(event_ids)
         return sorted(
-            events, key=lambda e: (-e.depth, -e.internal_metadata.stream_ordering)
+            events, key=lambda e: (-e.depth, -e.internal_metadata.stream_ordering)  # type: ignore[operator]
```

Comment:
Maybe worth a comment. @matrix-org/synapse-core how does this read?

```suggestion
            # type-ignore: mypy doesn't like negating the Optional[int] stream_ordering.
            # But it's never None, because these events were previously persisted to the DB.
            events, key=lambda e: (-e.depth, -e.internal_metadata.stream_ordering)  # type: ignore[operator]
```

Reply:
That sounds reasonable, yes.